skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/CodeOwnerWidget.kt

```diff
@@ -0,0 +1,142 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.sgp.intellij.ui
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent
+import com.intellij.openapi.fileEditor.FileEditorManagerListener
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.openapi.ui.popup.ListPopup
+import com.intellij.openapi.ui.popup.PopupStep
+import com.intellij.openapi.ui.popup.util.BaseListPopupStep
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.openapi.wm.StatusBarWidget
+import com.intellij.openapi.wm.impl.status.EditorBasedWidget
+import com.slack.sgp.intellij.codeowners.CodeOwnerFileFetcherImpl
+import com.slack.sgp.intellij.codeowners.CodeOwnerInfo
+import com.slack.sgp.intellij.codeowners.CodeOwnerRepository
+
+const val CODE_OWNER_WIDGET_ID = "CodeOwnerWidget"
+
+/** CodeOwnerWidget is responsible for displaying and handling code owner widget functionality. */
+class CodeOwnerWidget(project: Project) :
+  EditorBasedWidget(project), StatusBarWidget.MultipleTextValuesPresentation {
+
+  init {
+    // Connect to message bus and listen for file editor changes.
+    val bus = ApplicationManager.getApplication().messageBus
+    val connection = bus.connect(this)
+    connection.subscribe(
+      FileEditorManagerListener.FILE_EDITOR_MANAGER,
+      object : FileEditorManagerListener {
+        override fun selectionChanged(event: FileEditorManagerEvent) {
+          this@CodeOwnerWidget.selectionChanged(event)
+        }
+      }
+    )
+  }
+
+  private var currentSelectedFile: VirtualFile? = null
+  private val codeOwnerFileFetcher = CodeOwnerFileFetcherImpl(project)
+  private val codeOwnersRepo = CodeOwnerRepository(codeOwnerFileFetcher)
+  private val logger = logger<CodeOwnerWidget>()
+
+  override fun ID() = CODE_OWNER_WIDGET_ID
+
+  override fun getPresentation(): StatusBarWidget.WidgetPresentation = this
+
+  override fun selectionChanged(event: FileEditorManagerEvent) {
+    // New selected file is taken from a few sources in the following order:
+    // 1. newFile from FileEditorManagerEvent
+    // 2. selectedFile from EditorBasedWidget superclass
+    // 3. The previously selected file cached here
+    currentSelectedFile = event.newFile
```

Comment:
Does this mean we don't cache the selected file anymore? Also it looks like the comment above should be updated?

Reply:
yea correct, relying on the `event.newFile` seems much more reliable after some real-world use. Good catch on the comment, updated!